python/ray/train/tests/test_torch_trainer.py

```diff
@@ -111,6 +119,8 @@ def train_func():
     assert predictions.count() == 3
 
 
+# We can't really test for prepare_model here as we can't detect what the user
+# has saved without loading (and thus triggering the exception anyway)
```

Comment:
for my understanding, can you elaborate on why prepare_model causes this test to fail?

Reply:
`prepare_model` will wrap the model in DDP. If the user doesn't manually unwrap it before saving, DDP will throw an exception after being loaded.

Reply:
sorry I guess I mean more- why is it not going through the _encode_dict path?

Reply:
If a checkpoint is created from directory, we aren't really able to detect what's actually in the files without deserializing them in the first place (which would not only add overhead but also cause the error anyway).

Reply:
then for this dir checkpoint, why does it get deserialized in the first place?

Reply:
We can add prepare_model here but we'd have to unwrap the model before saving anyway, meaning we wouldn't really test anything extra here.